### DISCUSSION TITLE
Working with texts in slide masters

### DISCUSSION THREAD

**Discussion by bytehammer2** [ShapeCrawler/ShapeCrawler#276] [Ideas]
Awesome library!
I just discovered it yesterday and was excited and started using it right away.

One feature I need for a current project is changing the text and color in textboxes in slide masters programmatically.
Could you please implement this feature as soon as possible?

Thanks in advanced.

**Comment by AdamShakhabov** [comment]
Thank you for the feedback👍

The next release is scheduled for 15.09.2022. Will try to implement these features (https://github.com/ShapeCrawler/ShapeCrawler/issues/37)

**Reply by AdamShakhabov** [reply]
Done. @bytehammer2, please check [**v0.32.0**](https://github.com/ShapeCrawler/ShapeCrawler/releases/tag/v0.32.0)

**Reply by bytehammer2** [reply]
THANK YOU!!
I look forward to your updates and every library release.  Keep up the great work and thank you for sharing!!

**Comment by bytehammer2** [comment]
I'm revisiting my original request "Working with texts in slide masters" which you graciously implemented in v0.32 on Sep 9, 2022, again.
FYI, I always look forward with anticipation to your ShapeCrawler updates, but when I tried to update one of my projects using v0.49 to v0.50, I was surprised and somewhat disappointed to discover that this feature is no longer functional/unimplemented.  I also discovered that some of ShapeCrawler's code base has been refactored, e.g., IAutoShape to IShape, etc., which is understandable and was easily updated.  

Could you please consider reimplementing this feature as soon as possible?

**Reply by AdamShakhabov** [reply]
Could you please share the code that doesn't work for you? For instance, I was able to update the content and the color of a shape located on the Slide Master using the following code:

```c#
var pres = new Presentation(@"c:\test.pptx");
var shape = pres.SlideMasters[0].Shapes.First(sp => sp.Name == "Shape 1");

shape.TextFrame.Text = "edited";
shape.TextFrame.Paragraphs.First().Portions.First().Font.Color.Update("#008000");

pres.Save();
```

**Comment by bytehammer2** [comment]
Please accept my apologies, I found the issue that was causing the "Not Implemented" exception when trying to set the text in the placeholder textbox in the slide master.  It turns out the textbox's Text Options was set to "Resize shape to fit text" which caused the issue.  Once I set the textbox's option to "Do not Autofit" the exception no longer occurred.

Thanks for your attention and keep up the great work!
